### DISCUSSION TITLE
Entity Update issue

### DISCUSSION THREAD

**Discussion by FFasna** [AlexaCRM/dynamics-webapi-toolkit#103] [Q&A]
I'm updating an entity via WebAPI and the request went through and the value has been updated. But, I'm receiving this error

_**AlexaCRM\WebAPI\OData\Client::update(): Argument #2 ($key) must be of type string, null given, called in C:\Users\*****\vendor\alexacrm\dynamics-webapi-toolkit\src\WebAPI\Client.php on line 511**_

```
 $client = \AlexaCRM\WebAPI\ClientFactory::createOnlineClient(
            'https://*****.crm.dynamics.com/',
            env('CRM_APPLICATION_ID'),
            env('CRM_APPLICATION_SECRET'),
        );

        $record = new \AlexaCRM\Xrm\Entity('ldg', $row['ldg_id']);
        $record['ldgnotes'] = 'testing update';
        $client->Update($record);
```

I tried dd and checked the entity, the entity is retrieved correctly and the request was processed and updated in the CRM too.

Can anyone help me to fix the issue, please?

**Comment by georged** [comment]
@FFasna 
the code is correct but names for the table and columns don't seem to make sense. It looks like a custom table but what is the customization prefix and what is the table logical name?

For example, if your prefix is `ldg` and table logical name is `foobar` then your code should look like following:

```php
 $record = new \AlexaCRM\Xrm\Entity('ldg_foobar', $row['ldg_foobarid']);
 $record['ldg_notes'] = 'testing update';
 $client->Update($record);
```
HTH

**Reply by FFasna** [reply]
yes, the names are with a prefix and I edited them intentionally here. thanks for the reply I found the issue as well. This function was called inside an Excel import and the import method was considered empty rows after executing the correct rows. That's why I got the above error, where there is no data. So, I placed a condition to execute this above method, only the row has data.
